lang/cs/Org.Apache.REEF.Client.Tests/YarnClientTests.cs

```diff
@@ -128,6 +132,139 @@ public void TestGetApplication()
+                                })
+                        }
+                    }
+                }
+            };
+
+            const string expectedJson = @"{""application-id"":""AnyApplicationId"",""application-name"":""AnyAPP""," +
+@"""Queue"":null,""Priority"":1,""am-container-spec"":{""local-resources"":{""Entry"":[{""Key"":""APPLICATIONWILLFAILBUTWEDONTCAREHERE""," +
+@"""Value"":{""Resource"":""Foo"",""Type"":1,""Visibility"":2,""Size"":0,""Timestamp"":0}}]},""Environment"":null," +
+@"""Commands"":{""Command"":""DONTCARE""},""service-data"":null,""Credentials"":null,""application-acls"":null}," +
+@"""unmanaged-am"":false,""max-app-attempts"":1,""resource"":{""memory"":500,""VCores"":1},""application-type"":""REEFTest""," +
+@"""keep-containers-across-application-attempts"":false,""application-tags"":null}";
+
+            var thisApplication = new Application
+            {
+                AllocatedMB = 100,
+                AmHostHttpAddress = "http://anyhttpaddress",
+                AmContainerLogs = "SomeLogs",
+                ApplicationType = "AnyYarnApplicationType",
+                State = State.FINISHED,
+                Name = "AnyApplicationName",
+                RunningContainers = 0
+            };
+
+            var response = Substitute.For<IRestResponse>();
+            response.Headers.Returns(new List<Parameter>
+            {
+                new Parameter
+                {
+                    Name = "Location",
+                    Value = "http://somelocation"
+                }
+            });
+            response.StatusCode.Returns(HttpStatusCode.Accepted);
+
+            restReqExecutor.ExecuteAsync(
+                Arg.Is<IRestRequest>(
+                    req =>
+                        req.Resource == "ws/v1/cluster/apps"
+                        && req.Method == Method.POST
+                        && req.JsonSerializer is RestJsonSerializer
+                        && req.Parameters.First().Name == "application/json"
+                        && (string)req.Parameters.First().Value == expectedJson),
+                anyUri,
+                CancellationToken.None).Returns(Task.FromResult(response));
+
+            restReqExecutor.ExecuteAsync<Application>(
+                Arg.Is<IRestRequest>(
+                    req =>
+                        req.Resource == "ws/v1/cluster/apps/" + applicationId
+                        && req.RootElement == "app"
+                        && req.Method == Method.GET),
+                anyUri,
+                CancellationToken.None).Returns(Task.FromResult(thisApplication));
+
+            var yarnClient = ctx.GetClient();
+            Application actualApplication =
+                yarnClient.SubmitApplicationAsync(anySubmitApplication).GetAwaiter().GetResult();
+
+            Assert.AreEqual(thisApplication, actualApplication);
+            urlProvider.Received(2).GetUrlAsync();
+        }
+
         private class TestContext
```

Comment:
Yes it can be in test initialize, but then I will need shared fields in the class to pass the references to the objects that I need for my test. That is not a good design and can lead to buggy tests, I would rather use this tight encapsulation. Can you please elaborate on what is wrong with this method?